Libraries/Blob/URL.js

```diff
@@ -43,13 +42,143 @@ if (BlobModule && typeof BlobModule.BLOB_URI_SCHEME === 'string') {
  *
  * ```xml
  * <resources>
- *   <string name="blob_provider_authority">your.app.package.blobs</string>
+ *   <string name='blob_provider_authority'>your.app.package.blobs</string>
  * </resources>
  * ```
  */
-class URL {
-  constructor() {
-    throw new Error('Creating URL objects is not supported yet.');
+
+// Small subset from whatwg-url: https://github.com/jsdom/whatwg-url/tree/master/lib
+// The reference code bloat comes from Unicode issues with URLs, so those won't work here.
+export class URLSearchParams {
+  searchParams = [];
+
+  constructor(params: any) {
+    if (typeof params === 'object') {
+      Object.keys(params).forEach(key => this.append(key, params[key]));
+    }
+  }
+
+  append(key: string, value: string) {
+    this.searchParams.push([key, value]);
+  }
+
+  get paramString() {
+    if (this.searchParams.length === 0) {
+      return '';
+    }
+    const last = this.searchParams.length - 1;
+    return this.searchParams.reduce((acc, curr, index) => {
+      return acc + curr.join('=') + (index === last ? '' : '&');
+    }, '');
+  }
+
+  [Symbol.iterator]() {
+    return this.searchParams[Symbol.iterator]();
+  }
+
+  toString() {
+    return this.paramString;
+  }
+}
+
+export class URL {
```

Comment:
From a quick look, the codebase doesn't seem to use ES6 export. Maybe do `exports.URL = URL` instead to keep it consistent

Reply:
I changed it due to a flow warning about mixed exports from before I disabled flow. Is there a reason ES6 export wasn't part of the general sweep to convert things to ES6 classes, etc?

Reply:
I don't know the reason. maybe someone from FB can clarify. but will be nice to keep it consistent with other code regardless of the reason